tests/integration/synapseclient/integration_test.py

```diff
@@ -1193,3 +1205,136 @@ def test_create_delete_team(syn: Synapse) -> None:
+
+        # WHEN I used the rest_post_async method to get the bundle
+        request = {
+            "includeEntity": True,
+        }
+        entity_bundle = await self.syn.rest_post_async(
+            uri=f"/entity/{entity.id}/bundle2",
+            body=json.dumps(request),
+        )
+
+        # THEN I expect the bundle to contain the entity
+        assert entity_bundle["entity"]["id"] == entity.id
+        assert entity_bundle["entity"]["name"] == entity.name
+        assert entity_bundle["entity"]["etag"] == entity.etag
+
+    @pytest.mark.asyncio
+    async def test_rest_put_async(self, project: Project) -> None:
+        # GIVEN A bogus file to upload to synapse
+        path = utils.make_bogus_uuid_file()
+        self.schedule_for_cleanup(path)
+        entity = File(
+            name=FILE_NAME_PREFIX + str(uuid.uuid4()),
+            path=path,
+            parentId=project["id"],
+            description=FILE_DESCRIPTION,
+        )
+
+        # AND I store the file in Synapse
+        entity = self.syn.store(entity)
+        self.schedule_for_cleanup(entity.id)
+
+        # AND I used the rest_post_async method to get the bundle
+        request = {
+            "includeEntity": True,
+        }
+        entity_bundle = await self.syn.rest_post_async(
+            uri=f"/entity/{entity.id}/bundle2",
+            body=json.dumps(request),
+        )
+
+        # WHEN I updated a field on the entity
+        assert entity_bundle["entity"]["description"] == FILE_DESCRIPTION
+        entity_bundle["entity"]["description"] = NEW_DESCRIPTION
+
+        # AND I PUT the updated entity back to Synapse
+        updated_entity_bundle = await self.syn.rest_put_async(
+            uri=f"/entity/{entity_bundle['entity']['id']}/bundle2",
+            body=json.dumps({"entity": entity_bundle["entity"]}),
+        )
+
+        # THEN I expect the updated entity to have been stored to synapse
+        assert updated_entity_bundle["entity"]["description"] == NEW_DESCRIPTION
+
+        # AND getting the entity again from synapse should return the updated entity
+        entity_bundle_copy = await self.syn.rest_post_async(
+            uri=f"/entity/{entity.id}/bundle2",
+            body=json.dumps(
+                {
+                    "includeEntity": True,
+                }
+            ),
+        )
+        assert entity_bundle_copy["entity"]["description"] == NEW_DESCRIPTION
+
+    @pytest.mark.asyncio
+    async def test_rest_delete_async(self, project: Project) -> None:
+        # GIVEN A bogus file to upload to synapse
+        path = utils.make_bogus_uuid_file()
+        self.schedule_for_cleanup(path)
+        entity = File(
+            name=FILE_NAME_PREFIX + str(uuid.uuid4()),
+            path=path,
+            parentId=project["id"],
+            description=FILE_DESCRIPTION,
+        )
+
+        # AND I store the file in Synapse
+        entity = self.syn.store(entity)
+        self.schedule_for_cleanup(entity.id)
```

Comment:
Does this entity need to be scheduled for cleanup if you delete it in the next step?